src/SLCore/Common/Helpers/ConnectionIdHelper.cs

```diff
@@ -33,6 +35,28 @@ internal class ConnectionIdHelper : IConnectionIdHelper
         private const string SonarQubePrefix = "sq|";
         private static readonly Uri SonarCloudUri = new Uri("https://sonarcloud.io");
 
+        public string GetConnectionIdFromUri(Uri uri, string organisation)
+        {
+            if (uri is null)
+            {
+                return null;
+            }
+
+            if (uri == SonarCloudUri)
+            {
+                if (!string.IsNullOrWhiteSpace(organisation))
+                {
+                    return SonarCloudPrefix + organisation;
+                }
```

Comment:
Maybe for a refactoring change in the next PR: 
This would be easier to understand like this:
```
return string.IsNullOrWhiteSpace(organization)
    ? null
    : SonarCloudPrefix + organization;
```
and then removing the least `return null`